libbeat/outputs/amqp/client.go

```diff
@@ -463,12 +464,18 @@ func (c *client) prepareEvent(codec codec.Codec, incoming eventTracker, now time
 	if err != nil {
 		return nil, fmt.Errorf("exchange select: %v", err)
 	}
+	if exchangeName == "" {
+		return nil, fmt.Errorf("exchange name is missing")
+	}
 	c.logger.Debugf("calculated exchange name: %v", exchangeName)
 
 	routingKey, err := c.routingKeySelector.Select(content)
 	if err != nil {
 		return nil, fmt.Errorf("routing key select: %v", err)
 	}
+	if routingKey == "" {
+		return nil, fmt.Errorf("routing key is missing")
+	}
```

Comment:
A blank routing key is valid under AMQP though. It might not be our use case but it probably can't be an error for a generic AMQP pipeline like this.

Reply:
Ah very good point. will update. Thanks @gwilym!